src/chain/candidate.rs

```diff
@@ -7,26 +7,36 @@
 // permissions and limitations relating to use of the SAFE Network Software.
 
 use super::OnlinePayload;
-use crate::{id::PublicId, utils::LogIdent, utils::XorTargetInterval};
+use crate::{
+    id::PublicId,
+    time::{Duration, Instant},
+    utils::LogIdent,
+    utils::XorTargetInterval,
+};
 use log::LogLevel;
 use std::collections::BTreeSet;
 
+/// Duration after which a candidate is considered as expired.
+pub const CANDIDATE_EXPIRED_TIMEOUT: Duration = Duration::from_secs(90);
+
 /// A candidate (if any) may be in different stages of the resource proof process.
 /// When we consensus to accept them for resource proof, move to `AcceptedForResourceProof`
 /// with the value all elder expect to get from `CandidateInfo`.
 /// If we consensus to refuse them: reset to None.
 /// If we consensus to accept them: move to ApprovedWaitingSectionInfo until we are available to
 /// accept a new candidate, at which point reset to None.
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone)]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
 pub enum Candidate {
     /// No-one is currently in the resource proof process.
     /// We can take on a new candidate on consensus of an `ExpectCandidate` event.
     None,
-    /// We accepted a candidate to perform resource proof in this section. We are waiting for
-    /// them to send their `CandidateInfo` before starting the actual resource proof.
-    AcceptedForResourceProof {
+    /// We accepted a candidate for joining this section. We are waiting for them to send their
+    /// `CandidateInfo`.
+    Accepted {
         target_interval: XorTargetInterval,
         old_public_id: PublicId,
+        timestamp: Instant,
```

Comment:
That does not seem right, this is for shared state stuff, cannot have a timestamp.
This is why there was the candidate in peer manager that is different: In peer manager this is the view of each node. Here it is the state shared by all nodes.

Comment apply to other place that use it.

Reply:
This would clear up with the replacement of the candidate/prover parts that the simpler BootStrap process will give us. I suggest we impl the bootstrap process in this PR to show the whole logic and allow us to remove temporary code to keep tests working.

Reply:
If it is going away soon, a TODO comment to say timestamp should not be in chain  is probably enough to address my comment.

Reply:
I tried to see if I can remove the local candidate state (`Chain::candidate` and `PeerManager::candidate`) but there is a problem with the relocation interval which needs to be stored somewhere (so the section members can verify the new id of the joining node fits withing the interval). And as we are storing it we also need a mechanism to expire it so non-responding candidate won't block another nodes that want to join. Because of this I'm going to move the expiration info from `Chain` back to `PeerManager` for now.